addons/web_editor/models/ir_attachment.py

```diff
@@ -71,6 +71,15 @@ def _compute_image_size(self):
                 attachment.image_width = 0
                 attachment.image_height = 0
 
+    def _get_serve_attachment(self, url, extra_domain=None, order=None):
+        # Bypass the default IrAttachment._search() behavior which adds
+        # ('res_field', '=', False) to the domain if no `res_field` nor `id` is
+        # set in the domain, as we at least want to fetch ('res_field', 'in',
+        # (False, 'binary_content')). We add a meaningless id domain in case
+        # some other module would have a legit 'res_field' query.
+        extra_domain = (extra_domain or []) + [('id', '!=', False)]
+        return super()._get_serve_attachment(url, extra_domain, order)
+
```

Comment:
Beware, this is not classic class inheritance ! This change will apply to all `ir.attachment` as soon as `web_editor` is installed... (aka always) And I doubt the other applications would expect this behavior to change.
Maybe try to condition this with a context key:
From caller:
```py
attachment.with_context(any_attachment=True).something()
```
In `_get_serve_attachment`:
```py
if self.env.context.get('any_attachment'):
    extra_domain = (extra_domain or []) + [('id', '!=', False)]
```

Reply:
It's a reasonable idea, but I don't see how to apply it. We get there through `_serve_fallback`, `_serve_db`... Meaning that adding the context to those in `web_editor` (or even in `web_unsplash`, because I think it's only a problem to serve unsplash images) poses the same issue.
Basically, this covers the `/unsplash/<unsplash_id>/<name>.<ext>` route, which isn't served by a dedicated route.

Reply:
Then maybe include this method in `web_unsplash`, and use the request's path in the condition (e.g. match id against a regex) ?